core/src/impl/Kokkos_GraphImpl.hpp

```diff
@@ -108,6 +109,39 @@ struct _add_graph_kernel_tag<PolicyTemplate<PolicyTraits...>> {
   using type = PolicyTemplate<PolicyTraits..., IsGraphKernelTag>;
 };
 
+// Helper for the 'then', such that the user can indeed pass a callable that
+// takes no argument.
+template <typename Functor>
+struct ThenWrapper {
+  Functor functor;
+  template <typename T>
+  KOKKOS_FUNCTION void operator()(const T) const {
+    functor();
+  }
+};
+
+template <typename ExecutionSpace, typename Functor>
+struct GraphNodeThenImpl
```

Comment:
This definition is probably not in its final resting place. Anyone has a better idea where to put this one ?

Reply:
extra file?

Reply:
It's now in `core/src/impl/Kokkos_GraphNodeThenImpl.hpp`.